src/metabase/pulse/render/js_svg.clj

```diff
@@ -170,7 +170,8 @@
   [value goal settings]
   (let [js-res (js/execute-fn-name (context) "progress"
                                    (json/generate-string {:value value :goal goal})
-                                   (json/generate-string settings))
+                                   (json/generate-string settings)
+                                   (json/generate-string (:colors settings)))
```

Comment:
@adam-james-v I'm sorry, I forgot to submit more comments yesterday (I always add comments in a batch and submit all of them in one go) 🤦 

I touched on this a bit as I wanted to make the code consistent with what we do now.

I traced back to this PR that introduce the key `:colors`
https://github.com/metabase/metabase/pull/18277/files#diff-79c051a13f9e859c56dc5bffd3f145bb696e2da951d6ff45dc1fa38d4a3b9807R292

To be fair, FE doesn't have `colors` key in the visualization settings for all charts (I'm not sure if certain charts has it but most likely not) since the whitelabel colors (appearance settings) can be grabbed from JS's global `Metabase` and when JS bootstrapped it uses dynamically replace the colors in https://github.com/metabase/metabase/blob/master/frontend/src/metabase/lib/colors/palette.ts at runtime.

That said, I think it's easier and less ambiguous to pass colors directly to this function as an argument, actually, I did that the first time but changed to this version for consistency.

What I did previously here
https://github.com/metabase/metabase/blob/8efab7f6257e10d01b9e616df493a1113784b4a0/src/metabase/pulse/render/body.clj#L505

is to pass the colors explicitly like so:
```clojure
(js-svg/progress value goal settings (public-settings/application-colors)))]
```

Do you guys think that would be better? If so, I could also do that with `combo-chart` function for consistency.

Reply:
> I wonder if the settings map can be spec'd reasonably? Maybe that's something for a release cycle in future?

yes and yes, imho.